recipes/lora_finetune_single_device.py

```diff
@@ -779,6 +788,38 @@ def train(self) -> None:
                     )
                 )
 
+                # VALIDATION LOOP
+                self._sampler_val.set_epoch(curr_epoch)
+
+                pbar = tqdm(total=len(self._dataloader_val))
+                val_losses = []
+                for idx, batch in enumerate(self._dataloader_val):
+                    utils.batch_to_device(batch, self._device)
+
+                    current_loss = self._loss_step(batch)
```

Comment:
I think we will need to toggle eval <-> train mode for the model, right? (Another reason having a separate method will probably be cleanest)

Reply:
Thanks @ebsmothers ! It makes sense. I will look into it. Do you have any pointers on how to do that?